chia/cmds/cmds_util.py

```diff
@@ -10,3 +27,131 @@ def transaction_submitted_msg(tx: TransactionRecord) -> str:
 
 def transaction_status_msg(fingerprint: int, tx_id: bytes32) -> str:
     return f"Run 'chia wallet get_transaction -f {fingerprint} -tx 0x{tx_id}' to get status"
+
+
+async def validate_client_connection(
+    rpc_client: RpcClient, node_type: str, rpc_port: int, fingerprint: Optional[int]
+) -> Tuple[Optional[RpcClient], Optional[int]]:
+    from aiohttp import ClientConnectorError
+
+    try:
+        await rpc_client.healthz()
+        if type(rpc_client) == WalletRpcClient:
+            wallet_and_f = await get_wallet(rpc_client, fingerprint=fingerprint)
+            if wallet_and_f is None:
+                return None, fingerprint
+            else:
+                rpc_client, fingerprint = wallet_and_f
+    except ClientConnectorError:
+        print(f"Connection error. Check if {node_type.replace('_', ' ')} rpc is running at {rpc_port}")
+        print(f"This is normal if {node_type.replace('_', ' ')} is still starting up")
+        return None, fingerprint
+    return rpc_client, fingerprint
+
+
+@asynccontextmanager
+async def get_any_service_client(
+    node_type: str,
+    rpc_port: Optional[int] = None,
+    root_path: Path = DEFAULT_ROOT_PATH,
+    fingerprint: Optional[int] = None,
+) -> AsyncIterator[Tuple[Optional[Any], Dict[str, Any], Optional[int]]]:
+    # in the return the dict is the config & the first element is the rpc client.
+    from chia.util.config import load_config
+    from chia.util.ints import uint16
+
+    if node_type not in NODE_TYPES.keys():
+        print(f"Invalid node type: {node_type}")
+        return
+    # load variables from config file
+    config = load_config(root_path, "config.yaml")
+    self_hostname = config["self_hostname"]
+    if rpc_port is None:
+        rpc_port = config[node_type]["rpc_port"]
+    # select node client type based on string
+    try:
+        node_client = await NODE_TYPES[node_type].create(self_hostname, uint16(rpc_port), root_path, config)
+        # check if we can connect to node, and if we can then validate fingerprint access, otherwise return None
+        o_node_client, fingerprint = await validate_client_connection(node_client, node_type, rpc_port, fingerprint)
+        yield o_node_client, config, fingerprint
+    finally:
+        if node_client is not None:
+            node_client.close()
+            await node_client.await_closed()
+
+
+async def get_wallet(
+    wallet_client: WalletRpcClient, fingerprint: Optional[int]
+) -> Optional[Tuple[WalletRpcClient, int]]:
```

Comment:
Tuple outputs are bad.  Why do we return the client that was passed to us?

Reply:
Fair point, i don't think this is needed.